### PR TITLE
4.13/master stabilisation PR

### PR DESCRIPTION
**Pull request by rhtyd**
Smoketests fixes. Refer health check PR https://github.com/apache/cloudstack/pull/3517

Known/Failing tests:

Test | Result | Time (s) | Test File
-- | -- | -- | --
test_reboot_router | Error | 349.14 | test_network.py
test_08_start_router | Error | 71.81 | test_routers.py
test_09_reboot_router | Error | 0.12 | test_routers.py
test_02_list_snapshots_with_removed_data_store | Error | 1.18 | test_snapshots.py
test_01_volume_usage | Error | 9.42 | test_usage.py

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by rhtyd** [comment]
With the NPE fixes, it should fix the following:
test_02_list_snapshots_with_removed_data_store | Error | 1.18 | test_snapshots.py
test_01_volume_usage | Error | 9.42 | test_usage.py

@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by rhtyd** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/vm/UserVmManagerImpl.java [hunk 1]
```diff
@@ -1473,7 +1473,7 @@ public UserVm updateDefaultNicForVirtualMachine(UpdateDefaultNicForVMCmd cmd) th
             UsageEventUtils.publishUsageEvent(EventTypes.EVENT_NETWORK_OFFERING_ASSIGN, vmInstance.getAccountId(), vmInstance.getDataCenterId(), vmInstance.getId(),
                     oldNicIdString, oldNetworkOfferingId, null, 0L, VirtualMachine.class.getName(), vmInstance.getUuid(), vmInstance.isDisplay());
 
-            if (vmInstance.getState() != State.Stopped) {
+            if (vmInstance.getState() == State.Running) {
```
Comment (anuragaw) [inline server/src/main/java/com/cloud/vm/UserVmManagerImpl.java]: Log warning perhaps if this method is called when not in running state?

Comment (rhtyd) [reply server/src/main/java/com/cloud/vm/UserVmManagerImpl.java]: I think the case should trigger only when user vm is in running state

**Pull request merged**